views/components/ColorPicker.jsx

```diff
@@ -18,22 +13,49 @@ const ColorPicker = ({
 
   return (
     <div className={classes.swatchContainer}>
-      <Typography variant='caption' className={classes.colorLabel}>
-          Color
+      <Typography variant="caption" className={classes.colorLabel}>
+        Color
       </Typography>
       <Paper
         className={classes.swatch}
         style={{ backgroundColor: color }}
         onClick={() => setColorPickerToggle(true)}
       />
-    {isColorPickerOpen && (
-      <div className={classes.popover} ref={popover}>
-        <HexColorPicker 
-          color={color} 
-          onChange={(newColor) => onColorChange({target: { name: colorKey, value: newColor }}, idx)} 
-        />
-      </div>
-    )}
+      {isColorPickerOpen && (
+        <div
+          className={classes.popover}
+          ref={popover}
+          key={idx + 'color_picker'}
+        >
+          <HexColorPicker
+            color={color}
+            onChange={(newColor) =>
+              onColorChange(
+                { target: { name: colorKey, value: newColor } },
+                idx
+              )
+            }
+          />
+          <div className={classes.colorPickerSwatchBox}>
+            {presetColors.map((presetColor, i) => (
+              <button
+                key={presetColor + i + 'swatch'}
```

Comment:
We could drop the index here as there should only be one occurrence of each hex in the list. Adding `'swatch'` is fine but unnecessary as it will be the same for each key, and is not used by the component it's passed to, it's just a [helper to React](https://reactjs.org/docs/lists-and-keys.html#keys) when rendering things via `map`.
```suggestion
            {presetColors.map((presetColor) => (
              <button
                key={presetColor}
```

Reply:
Thanks for sharing 👍

Reply:
https://github.com/AMP-SCZ/dpdash/pull/183/commits/575b52832ef21a014787ebbec9f099dcbd288fbd